supportedBackends/bcs/src/main/scala/cromwell/backend/impl/bcs/BcsAsyncBackendJobExecutionActor.scala

```diff
@@ -0,0 +1,361 @@
+  }
+
+  private def relativePath(path: String): Path = {
+    val absolutePath = DefaultPathBuilder.get(path) match {
+      case p if !p.isAbsolute => commandDirectory.resolve(p)
+      case p => p
+    }
+
+    absolutePath
+  }
+
+  private[bcs] lazy val callRawOutputFiles: List[WomFile] = {
+    import cats.syntax.validated._
+    def evaluateFiles(output: OutputDefinition): List[WomFile] = {
+      Try (
+        output.expression.evaluateFiles(jobDescriptor.localInputs, NoIoFunctionSet, output.womType).map(_.toList)
+      ).getOrElse(List.empty[WomFile].validNel)
+        .getOrElse(List.empty)
+    }
+
+    // val womFileOutputs = call.task.findOutputFiles(jobDescriptor.fullyQualifiedInputs, PureStandardLibraryFunctions)
+
+    jobDescriptor.taskCall.callable.outputs.flatMap(evaluateFiles)
+  }
+
+  private[bcs] def isOutputOssFileString(s: String): Boolean = {
+    println(callRawOutputFiles)
+    callRawOutputFiles.exists({
+      case file: WomSingleFile if file.value == s => true
+      case _ => false
+    })
+  }
+
+  private[bcs] def generateBcsOutputs(jobDescriptor: BackendJobDescriptor): Seq[BcsMount] = {
+    val outputs = callRawOutputFiles.distinct flatMap { womFile =>
+      womFile match {
+        case singleFile: WomSingleFile => List(generateBcsSingleFileOutput(singleFile))
+        case _: WomGlobFile => throw new RuntimeException(s"glob output not supported currently")
+      }
+    }
+
+    outputs.toSeq
+  }
+
+  private def generateBcsSingleFileOutput(wdlFile: WomSingleFile): BcsOutputMount = {
+    val destination = getPath(wdlFile.valueString) match {
+      case Success(ossPath: OssPath) => ossPath
+      case _ => callRootPath.resolve(wdlFile.value.stripPrefix("/"))
+    }
+
+    val src = relativePath(mapCommandLineWomFile(wdlFile).valueString)
+
+    BcsOutputMount(src, destination, false)
+  }
+
+  private[bcs] def getOssFileName(ossPath: OssPath): String = {
+    getPath(ossPath.pathWithoutScheme) match {
+      case Success(path) => path.getFileName.pathAsString
+      case _ => ossPath.pathWithoutScheme
+    }
+  }
+
+  private[bcs] def localizeOssPath(ossPath: OssPath): String = {
+    if (isOutputOssFileString(ossPath.pathAsString)) {
+      commandDirectory.resolve(getOssFileName(ossPath)).pathAsString
+    } else {
+      userDefinedMounts collectFirst {
+        case bcsMount: BcsMount if ossPath.pathAsString.startsWith(bcsMount.src.pathAsString) =>
+          bcsMount.dest.resolve(ossPath.pathAsString.stripPrefix(bcsMount.src.pathAsString)).pathAsString
+      } getOrElse {
+        val mount = ossPathToMount(ossPath)
+        mount.dest.pathAsString
+      }
+    }
+
+  }
+
+  override def mapCommandLineWomFile(womFile: WomFile): WomFile = {
+    getPath(womFile.valueString) match {
+      case Success(ossPath: OssPath) =>
+        WomFile(WomSingleFileType, localizeOssPath(ossPath))
+      case _ => womFile
+    }
+  }
+
+
+  override lazy val commandLineValueMapper: WomValue => WomValue = {
```

Comment:
@crisish Can you explain more why the override is necessary? I know you put in some WDL above but I'm not familiar enough w/ OSS to know what's going.

Reply:
That example WDL above is an interesting use case. I don't know enough about OSS nor the new BCS backend at the moment, but I would never tell a WDL author to pass a path that way.

In ~`JES`~ `PAPI` (nor in the `Local` backend really) output paths are (I believe?) assumed to be relative. If a user tries to specify an absolute path like this they're going to have a bad time.

This WDL wouldn't be portable, and may not be 100% supported/recommended as it might try to overwrite the absolute path on a second run:
```wdl
workflow three_step {
  call ps {
    input: out = "gs://my-hard-coded-bucket/top/level/path/to/file"
  }
}
```

Instead I would tell the author to do:
```wdl
workflow three_step {
  call ps {
    input: out = "path/to/file"
  }
}
```